x/sequencer/keeper/proposer.go

```diff
@@ -37,86 +36,47 @@ func (k Keeper) RollappPotentialProposers(ctx sdk.Context, rollappId string) []t
 	return append(seqs, k.SentinelSequencer(ctx))
 }
 
-// UpdateProposerIfNeeded will assign a proposer to the rollapp. It won't replace the incumbent proposer
-// if they are not sentinel. Otherwise it will prioritise a non sentinel successor. Finally, it
+// RecoverFromSentinelProposerIfNeeded will assign a proposer to the rollapp. It won't replace the incumbent proposer
+// if they are not sentinel. Otherwise it will prioritize a non sentinel successor. Finally, it
 // choose one based on an algorithm.
 // The result can be the sentinel sequencer.
-func (k Keeper) UpdateProposerIfNeeded(ctx sdk.Context, rollapp string) error {
+func (k Keeper) RecoverFromSentinelProposerIfNeeded(ctx sdk.Context, rollapp string) error {
 	proposer := k.GetProposer(ctx, rollapp)
-	before := proposer
 
 	// a valid proposer is already set so there's no need to do anything
 	if !proposer.Sentinel() {
 		return nil
 	}
+
+	before := proposer
 	successor := k.GetSuccessor(ctx, rollapp)
-	k.SetProposer(ctx, rollapp, successor.Address)
-	k.SetSuccessor(ctx, rollapp, types.SentinelSeqAddr)
-	if k.GetProposer(ctx, rollapp).Sentinel() {
+	// if successor is sentinel, we attempt to find a non sentinel successor
```

Comment:
at what point will the successor be not sentinel, if we only reached this part if the proposer is sentinel?

Reply:
yeah not sure
refactored